salt/modules/zypperpkg.py

```diff
@@ -218,6 +222,22 @@ def stderr(self):
     def pid(self):
         return self.__call_result.get('pid', '')
 
+    def calc_zypper(self):
+        try:
+            zypp_tmp = version('zypper')
+            # zypper version 1.11.34 in SLE12 update supports vendor change for only dist upgrade
+            if version_cmp(zypp_tmp, '1.11.34') >= 0 and version_cmp(zypp_tmp, '1.14.8') == -1:
```

Comment:
This code is doing some weird things with the flags.

For example, let assume the Zypper version is `1.15`. It should support "vendorchange" flags for everything (dup, install, patch..), but your flags would be set as:

```
dup_avc = False
inst_avc = True
```

And this is kind of inconsistent. Both flags should be `True` in that case. I'm not sure if I'm missing something.

I would remove this part of the version checking, to have consistency between the two flags

```suggestion
            if version_cmp(zypp_tmp, '1.11.34') >= 0:
```

Does that make sense for you?

Reply:
Yes, this is good thanks.